uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequest.php

```diff
@@ -139,14 +138,28 @@ protected function update($params) {
    */
   private function getErrorMessageForInvalidStatusTransition($params) {
     $leaveStatuses = LeaveRequest::getStatuses();
-    $isOwnRequest = CRM_Core_Session::getLoggedInContactID() === $params['contact_id'];
+    $isOwnRequest = $this->leaveRequestRightsService->currentUserIsLeaveContact($params['contact_id']);
 
     if ($isOwnRequest && (int)$params['status_id'] === (int)$leaveStatuses['approved']) {
-      return "You can't approve your own leave requests";
+      return 'You can\'t approve your own leave requests';
     }
 
-    return "You can't change the Leave Request status from " .
-      $this->getCurrentStatus($params) . " to {$params['status_id']}";
+    return 'You can\'t change the Leave Request status from "' .
+      $this->getLeaveStatusReadableLabel($this->getCurrentStatus($params)) . '" to "' .
```

Comment:
Instead of writing a custom function, `LeaveRequest::buildOptions('status_id')` will return an array of status labels indexed by the status id.

Reply:
Done.